File: backend-ms1/src/main/java/ma/zs/easystock/service/facade/admin/commun/EtatCommandeAdminService.java
package ma.zs.easystock.service.facade.admin.commun;

import java.util.List;
import ma.zs.easystock.bean.core.commun.EtatCommande;
import ma.zs.easystock.dao.criteria.core.commun.EtatCommandeCriteria;
import ma.zs.easystock.zynerator.service.IService;

import ma.zs.easystock.ws.dto.commun.EtatCommandeDto;
import org.springframework.http.HttpEntity;


public interface EtatCommandeAdminService extends  IService<EtatCommande,EtatCommandeCriteria>  {



    HttpEntity<byte[]> createPdf(EtatCommandeDto dto) throws Exception;

}

File: frontend/src/app/controller/service/admin/commun/EtatCommandeAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {EtatCommandeDto} from 'src/app/controller/model/commun/EtatCommande.model';
import {EtatCommandeCriteria} from 'src/app/controller/criteria/commun/EtatCommandeCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class EtatCommandeAdminService extends AbstractService<EtatCommandeDto, EtatCommandeCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/etatCommande/');
    }

    public constrcutDto(): EtatCommandeDto {
        return new EtatCommandeDto();
    }

    public constrcutCriteria(): EtatCommandeCriteria {
        return new EtatCommandeCriteria();
    }
}
